### PR TITLE
[Blaze] Allow setting a CTA text when creating a campaign

### LINKED ISSUE #14371: [Blaze] Update Networking and Storage to include the `cta_text` field

### LINKED ISSUE #14372: [Blaze] Update ad creation to allow editing the CTA text

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #14371 
Closes: #14372 

## Description
This PR adds support for setting a CTA text on the created campaign, it updates the following:
- Adds a new field `ctaText` to the networking model.
- Updates the Ad editor screen to allow changing the CTA text value.
- Updates the Ad preview to highlight the selected CTA text, the preview is also updated to match the web design to be a more accurate representation of how the ad would look.

For now, the CTA text is empty by default, in my next PR, I'll add logic to use AI to pre-fill it like the tagline and description.

## Steps to reproduce
1. Use a site eligible for Blaze.
2. Open the app.
3. Start creating a campaign.

## Testing information
- Confirm the CTA text updates are reflected on the preview.
- Confirm how the preview looks when the CTA text is empty.
- Create a campaign, and confirm the correct CTA text in the campaign details.

## Screenshots

https://github.com/user-attachments/assets/55448d26-6bc4-4ec2-91a9-5fe0804811ab

**Web preview**
| With CTA text | Without CTA Text |
| --- | --- |
| Screenshot 2024-11-12 at 16 02 04 | Screenshot 2024-11-12 at 16 02 17 |

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationForm.swift [hunk 1]
```diff
@@ -415,13 +429,21 @@ private extension BlazeCampaignCreationForm {
...
+        static let ctaButtonPadding = EdgeInsets(top: 10, leading: 16, bottom: 10, trailing: 16)
+        static let ctaButtonTopMargin: CGFloat = 4
     }
 
     enum Constants {
         static let backgroundViewColor = Color(light: .init(uiColor: .systemBackground),
                                                dark: .init(uiColor: .secondarySystemBackground))
         static let cellColor = Color(light: .init(uiColor: .systemBackground),
                                      dark: .init(uiColor: .tertiarySystemBackground))
+        static let ctaButtonColor = Color(red: 0, green: 0.219, blue: 1)
```
Comment (hichamboushaba) [inline WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationForm.swift:442]: This color matches what the web client uses.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/232">21.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by hafizrahman — APPROVED** [review]
I tested this on Simulator with iPhone 15/iOS 18.0 and iPad Pro 11 inch/iOS 17.5 and the UI and validation for CTA work. Empty and non-empty CTA case are handled correctly.

Code-wise, it looks good to me too. Thanks for re-using existing/similar code both for the CTA field in the View and its various validation and error message display in the view model, it made things easier to review.

Non-blocking, but I think this can be mentioned in release note as it's a new feature addition. Please feel free to merge after adding it 👍🏼

**Comment by hichamboushaba** [comment]
Thanks @hafizrahman for the review

> Non-blocking, but I think this can be mentioned in release note as it's a new feature addition. Please feel free to merge after adding it 👍🏼

I already added the release note in the second PR of the change https://github.com/woocommerce/woocommerce-ios/pull/14399, so I'll proceed with merging this one.

**Pull request merged**